services/libs/tracing/src/tracer.ts

```diff
@@ -0,0 +1,83 @@
+import fs from 'fs'
+
+import opentelemetry, { Tracer } from '@opentelemetry/api'
+import { NodeSDK } from '@opentelemetry/sdk-node'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc'
+import { Resource, ResourceAttributes } from '@opentelemetry/resources'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express'
+import { AwsInstrumentation } from '@opentelemetry/instrumentation-aws-sdk'
+import { SequelizeInstrumentation } from 'opentelemetry-instrumentation-sequelize'
+
+let sdk: NodeSDK | undefined
+let isInitialized = false
+
+export const getServiceTracer = (): Tracer => {
+  const service = process.env['SERVICE']
+  if (!service) {
+    throw new Error('Environment variable `SERVICE` is not set')
+  }
+
+  if (isInitialized) {
+    return opentelemetry.trace.getTracer(service)
+  }
+
+  const attrs: ResourceAttributes = {
+    [SemanticResourceAttributes.SERVICE_NAME]: service,
+  }
+
+  // Try to retrieve the Kubernetes namespace from file. If found, we can assume
+  // service is running in Kubernetes so we can set other Kubernetes-related
+  // resources such as pod and deployment names.
+  try {
+    const data = fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/namespace')
```

Comment:
Good idea, but we have an env variable `KUBE_MODE` specifically for this, take a look

Reply:
Still, we need access to the namespace's name.

Reply:
Then how about we get this information in a proper way like this:
https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/

Because the pod name you're getting is not really a pod name. And the deployment name is based on a convention, but can get a real one from kubernetes.